en/embedding.html

```diff
@@ -17,58 +17,141 @@
 <h2 id="provided-embedders">Provided embedders</h2>
 
 <p>Vespa provides some embedders as part of the platform. Native Vespa embedders reduces complexity as developers does not need to 
-  host additional infrastructure for producing embeddings. Using the Vespa embedder functionality can also 
-  reduce latency as embeddings are large and network related latency can easily be higher than the ML computations.  
+  host additional infrastructure for producing text embeddings. Native Vespa embedders embeds text for both queries and documents.
+</p>
+<p>
+  An example of a Vespa <a href="query-api.html#http">query</a> request using Vespa embed functionality:
+<pre>{% highlight json %}
+ {
+    "yql": "select * from doc where {targetHits:10)nearestNeighbor(embedding,query_embedding)",
+    "query": "semantic search", 
+    "input.query(query_embedding)": "embed(semantic search)", 
+}
+{% endhighlight %}</pre>
+See more usage examples in <a href="#embedding-a-query-text">embedding a query text</a> and <a href="#embedding-a-document-field">embedding a document field</a>.
 </p>
 
+<h3 id="huggingface-embedder">Huggingface Embedder</h3>
 
-<h3 id="bertbase-embedder">BertBase embedder</h3>
+<p>An embedder using any <a href="https://huggingface.co/docs/tokenizers/index">Huggingface tokenizer</a>, including multilingual tokenizers,
+  to produce tokens which is then input to a supplied transformer model in ONNX model format.  
+</p>
+<p>This provides embeddings directly suitable for <a href="approximate-nn-hnsw.html">retrieval</a>
+  and <a href="ranking.html">ranking</a> in Vespa, and makes it easy
+  to implement semantic search with no need for custom components or client-side embedding inference
+  when used with the syntax for invoking the embedder in queries and during document indexing described
+  in <a href="#embedding-a-query-text">embedding a query text</a> and <a href="#embedding-a-document-field">embedding a document field</a>.</p>
+
+<p>The Huggingface embedder is configured in <a href="reference/services.html">services.xml</a>, within the <code>container</code> tag:</p>
 
-<p>An embedder using <a href="#wordpiece-embedder">WordPiece</a> to produce tokens which is then input to
-a supplied ONNX model on the form expected by a BERT base model. 
-See <a href="https://github.com/vespa-engine/sample-apps/blob/master/simple-semantic-search/export_model_from_hf.py">export_model_from_hf.py</a>,
-for how to export a <a href="https://www.sbert.net/">sentence-transformer</a> model to ONNX format.
-See also <a href="#troubleshooting-model-signature">troubleshooting model signature</a>.
+<pre>{% highlight xml %}
+<container id="default" version="1.0">
+    <component id="hf-embedder" type="hugging-face-embedder">
+        <transformer-model path="my-models/model.onnx"/>
+        <tokenizer-model path="my-models/tokenizer.json"/>
```

Comment:
Consider adding `max-tokens` to example as it's likely required.

Reply:
It is https://github.com/vespa-engine/vespa/blob/d7568f0d450df2287657ac18c37955a1867496f5/configdefinitions/src/vespa/hugging-face-embedder.def#L10